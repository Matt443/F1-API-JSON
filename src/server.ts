import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}