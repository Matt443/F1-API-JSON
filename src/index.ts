import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
