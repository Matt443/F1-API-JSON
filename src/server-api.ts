import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
