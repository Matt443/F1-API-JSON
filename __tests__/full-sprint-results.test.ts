import { getFullSprintResults } from "../src/server";

test("get race results", async () => {
    expect(await getFullSprintResults(2024)).toMatchSnapshot();
});
