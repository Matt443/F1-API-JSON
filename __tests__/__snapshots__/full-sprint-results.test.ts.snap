// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get race results 1`] = `
[
  {
    "code": "NOR",
    "laps": 24,
    "name": "Lando Norris",
    "number": 4,
    "points": 8,
    "team": "McLaren Mercedes",
    "time": "29:46.045",
  },
  {
    "code": "PIA",
    "laps": 24,
    "name": "Oscar Piastri",
    "number": 81,
    "points": 7,
    "team": "McLaren Mercedes",
    "time": "+0.593s",
  },
  {
    "code": "LEC",
    "laps": 24,
    "name": "Charles Leclerc",
    "number": 16,
    "points": 6,
    "team": "Ferrari",
    "time": "+5.656s",
  },
  {
    "code": "VER",
    "laps": 24,
    "name": "Max Verstappen",
    "number": 1,
    "points": 5,
    "team": "Red Bull Racing Honda RBPT",
    "time": "+6.497s",
  },
  {
    "code": "SAI",
    "laps": 24,
    "name": "Carlos Sainz",
    "number": 55,
    "points": 4,
    "team": "Ferrari",
    "time": "+7.224s",
  },
  {
    "code": "RUS",
    "laps": 24,
    "name": "George Russell",
    "number": 63,
    "points": 3,
    "team": "Mercedes",
    "time": "+12.475s",
  },
  {
    "code": "GAS",
    "laps": 24,
    "name": "Pierre Gasly",
    "number": 10,
    "points": 2,
    "team": "Alpine Renault",
    "time": "+18.161s",
  },
  {
    "code": "PER",
    "laps": 24,
    "name": "Sergio Perez",
    "number": 11,
    "points": 1,
    "team": "Red Bull Racing Honda RBPT",
    "time": "+18.717s",
  },
  {
    "code": "LAW",
    "laps": 24,
    "name": "Liam Lawson",
    "number": 30,
    "points": 0,
    "team": "RB Honda RBPT",
    "time": "+20.773s",
  },
  {
    "code": "ALB",
    "laps": 24,
    "name": "Alexander Albon",
    "number": 23,
    "points": 0,
    "team": "Williams Mercedes",
    "time": "+24.606s",
  },
  {
    "code": "HAM",
    "laps": 24,
    "name": "Lewis Hamilton",
    "number": 44,
    "points": 0,
    "team": "Mercedes",
    "time": "+29.764s",
  },
  {
    "code": "COL",
    "laps": 24,
    "name": "Franco Colapinto",
    "number": 43,
    "points": 0,
    "team": "Williams Mercedes",
    "time": "+33.233s",
  },
  {
    "code": "OCO",
    "laps": 24,
    "name": "Esteban Ocon",
    "number": 31,
    "points": 0,
    "team": "Alpine Renault",
    "time": "+34.128s",
  },
  {
    "code": "BEA",
    "laps": 24,
    "name": "Oliver Bearman",
    "number": 50,
    "points": 0,
    "team": "Haas Ferrari",
    "time": "+35.507s",
  },
  {
    "code": "TSU",
    "laps": 24,
    "name": "Yuki Tsunoda",
    "number": 22,
    "points": 0,
    "team": "RB Honda RBPT",
    "time": "+41.374s",
  },
  {
    "code": "BOT",
    "laps": 24,
    "name": "Valtteri Bottas",
    "number": 77,
    "points": 0,
    "team": "Kick Sauber Ferrari",
    "time": "+43.231s",
  },
  {
    "code": "ZHO",
    "laps": 24,
    "name": "Zhou Guanyu",
    "number": 24,
    "points": 0,
    "team": "Kick Sauber Ferrari",
    "time": "+54.139s",
  },
  {
    "code": "ALO",
    "laps": 24,
    "name": "Fernando Alonso",
    "number": 14,
    "points": 0,
    "team": "Aston Martin Aramco Mercedes",
    "time": "+56.537s",
  },
  {
    "code": "STR",
    "laps": 24,
    "name": "Lance Stroll",
    "number": 18,
    "points": 0,
    "team": "Aston Martin Aramco Mercedes",
    "time": "+57.983s",
  },
  {
    "code": "HUL",
    "laps": 19,
    "name": "Nico Hulkenberg",
    "number": 27,
    "points": 0,
    "team": "Haas Ferrari",
    "time": "DNF",
  },
]
`;
